src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -246,194 +254,133 @@ std::shared_ptr<weighted_tree_node_t> get_sub_banks (
-
-                return nullptr;
-            }
-            rc = sqlite3_reset (b_select_associations_stmt);
-            if (rc != SQLITE_OK) {
-                std::cerr << sqlite3_errmsg (DB) << std::endl;
-                sqlite3_close (DB);
-
-                return nullptr;
-            }
-            rc = sqlite3_clear_bindings (b_select_sub_banks_stmt);
-            if (rc != SQLITE_OK) {
-                std::cerr << sqlite3_errmsg (DB) << std::endl;
-                sqlite3_close (DB);
-
-                return nullptr;
-            }
-            rc = sqlite3_reset (b_select_sub_banks_stmt);
-            if (rc != SQLITE_OK) {
-                std::cerr << sqlite3_errmsg (DB) << std::endl;
-                sqlite3_close (DB);
-
-                return nullptr;
-            }
-            rc = sqlite3_clear_bindings (b_select_shares_stmt);
-            if (rc != SQLITE_OK) {
-                std::cerr << sqlite3_errmsg (DB) << std::endl;
-                sqlite3_close (DB);
-
-                return nullptr;
-            }
-            rc = sqlite3_reset (b_select_shares_stmt);
-            if (rc != SQLITE_OK) {
-                std::cerr << sqlite3_errmsg (DB) << std::endl;
-                sqlite3_close (DB);
-
+            rc = reset_and_clear_bindings (DB, c_assoc, c_sub_banks, c_shrs);
+            if (rc != SQLITE_OK)
                 return nullptr;
-            }
 
             if (get_sub_banks (DB,
                                b,
                                parent_bank,
-                               b_select_shares_stmt,
-                               b_select_sub_banks_stmt,
-                               b_select_associations_stmt) == nullptr) {
+                               c_shrs,
+                               c_sub_banks,
+                               c_assoc) == nullptr) {
                 std::cerr << "get_sub_banks () returned a nullptr" << std::endl;
                 return nullptr;
             }
         }
     }
-    return root_ptr;
+    return node;
 }
 
 std::shared_ptr<weighted_tree_node_t> load_accounting_db (
                                                     const std::string &path)
 {
     // SQL statements to retrieve data from flux-accounting database
-    sqlite3_stmt *b_select_root_bank_stmt = nullptr;
-    sqlite3_stmt *b_select_shares_stmt = nullptr;
-    sqlite3_stmt *b_select_sub_banks_stmt = nullptr;
-    sqlite3_stmt *b_select_associations_stmt = nullptr;
+    std::string s_shrs, s_sub_banks, s_assoc, s_root_bank;
+    sqlite3_stmt *c_root_bank = nullptr;
+    sqlite3_stmt *c_shrs = nullptr;
+    sqlite3_stmt *c_sub_banks = nullptr;
+    sqlite3_stmt *c_assoc = nullptr;
 
     sqlite3 *DB = nullptr;
     int rc = 0;
 
-    // open FluxAccounting DB in read-write mode; if it does not exist yet,
-    // create a new database file
+    std::string root_bank;
+    std::shared_ptr<weighted_tree_node_t> root = nullptr;
+
+    // open flux-accounting DB in read-write mode
```

Comment:
Could you check if `sqlite3_` functions set `errno` and if not set `errno` at this level. This will be useful when this function gets into a public API. `errno= EIO` makes sense to me but other may have a better idea.

Reply:
Thanks for clarifying this during our team meeting. I think I understand what you are suggesting - setting something like `errno = EIO` in case the SQLite database cannot be opened. I can push up a `fixup` commit that sets this.

Reply:
@garlick or @SteVwonder: Is there any other "catch-all" errno which works better with sqlite function failures like this?